pkg/cmd/join/exec.go

```diff
@@ -114,7 +117,33 @@ func (o *Options) run() error {
 		return err
 	}
 	output = append(output, out...)
-	fmt.Printf("Deploying klusterlet agent. Please wait a few minutes then log onto the hub cluster and run the following command:\n\n"+
+
+	timeout := int64(300)
+	watch, err := kubeClient.CoreV1().Pods("open-cluster-management-agent").Watch(context.TODO(), metav1.ListOptions{LabelSelector: "app=klusterlet-registration-agent", TimeoutSeconds: &timeout})
```

Comment:
I think you would want to check the status of klusterlet instead. There is a list of conditions in the klusterlet that you could use to check if it is working correctly

Reply:
Hi @qiujian16 ,
Per my understanding, the prerequisite of doing `accept` from hub cluster is the CSR is created from the hub cluster. Which should be done by the `klusterlet-registration-agent` from the spoke cluster.
So I'm watching it's status other than the `klusterlet` pod status.

Please correct me if I'm wrong. Thanks.

Reply:
oh, it is not the pod status. We have a `Klusterlet` CR on the managed cluster. And the status condition of this CR should report the status you would like to know.

Reply:
> oh, it is not the pod status. We have a `Klusterlet` CR on the managed cluster. And the status condition of this CR should report the status you would like to know.

Thanks for the info, let me update the code then.

Reply:
Hi @qiujian16 ,
I have tried to watch the status.condition of CR `klusterlet`, but unfortunately, even the latest condition shows `{Applied True 0 2021-09-17 15:54:31 +0800 CST KlusterletApplied Klusterlet Component Applied}`, user still cannot do `clusteradm accept --clusters kind-mc1` from hub cluster. As in this moment, the pod `klusterlet-registration-agent` is still in status `ContainerCreating`, the CSR has not been created from hub cluster.

I think there supposed to have a condition called `Available` for CR `klusterlet`, but I do not find it from my test env after a long time when the `klusterlet` has been created.
```
status:
  conditions:
  - lastTransitionTime: "2021-09-17T08:09:11Z"
    message: Failed to get kubeconfig from `kubectl get secret -n "open-cluster-management-agent"
      "hub-kubeconfig-secret" -ojsonpath='{.data.kubeconfig}'`. This is set by the
      klusterlet registration deployment, but the CSR must be approved by the cluster-admin
      on the hub.
    reason: HubKubeConfigMissing
    status: "True"
    type: KlusterletRegistrationDegraded
  - lastTransitionTime: "2021-09-17T08:09:11Z"
    message: Failed to get kubeconfig from `kubectl get secret -n "open-cluster-management-agent"
      "hub-kubeconfig-secret" -ojsonpath='{.data.kubeconfig}'`. This is set by the
      klusterlet registration deployment, but the CSR must be approved by the cluster-admin
      on the hub.
    reason: HubKubeConfigMissing
    status: "True"
    type: KlusterletWorkDegraded
  - lastTransitionTime: "2021-09-17T08:09:13Z"
    message: Klusterlet Component Applied
    reason: KlusterletApplied
    status: "True"
    type: Applied
  generations:
  - group: apps
    lastGeneration: 1
    name: klusterlet-registration-agent
    namespace: open-cluster-management-agent
    resource: deployments
    version: v1
  - group: apps
    lastGeneration: 1
    name: klusterlet-work-agent
    namespace: open-cluster-management-agent
    resource: deployments
    version: v1
  observedGeneration: 1
```

Reply:
I think the CR `klusterlet` is designed to have a status.condition called `Available`, but with unknown reason, there isn’t.
```
	// Conditions contain the different condition statuses for this Klusterlet.
	// Valid condition types are:
	// Applied: Components have been applied in the managed cluster.
	// Available: Components in the managed cluster are available and ready to serve.
	// Progressing: Components in the managed cluster are in a transitioning state.
	// Degraded: Components in the managed cluster do not match the desired configuration and only provide
	// degraded service.
	Conditions []metav1.Condition `json:"conditions"`
```

Reply:
that sounds like an enhancement we should make on the registration-operator.

Reply:
Let me enhance the `klusterletStatusController` to check all the pods runs on spoke cluster within namespace `open-cluster-management` and `open-cluster-management-agent`, if they are all in running status, then make the CR `klusterlet` in `Available` status.